auslib/db.py

```diff
@@ -2295,10 +2295,21 @@ def assertOptionsExist(self, permission, options):
                 raise ValueError('Unknown option "%s" for permission "%s"' % (opt, permission))
 
     def getAllUsers(self, transaction=None):
-        res_permissions = self.select(columns=[self.username], distinct=True, transaction=transaction)
-        res_roles = self.user_roles.select(columns=[self.user_roles.username], transaction=transaction)
-        res = res_roles + res_permissions
-        return list(set([r['username'] for r in res]))
+        res_users = self.select(columns=[self.username], distinct=True, transaction=transaction)
+        users_list = list(set([r['username'] for r in res_users]))
+        users = []
+        for user in users_list:
+            res_roles = self.user_roles.select(where=[
+                self.user_roles.username == user],
+                columns=[self.user_roles.role],
+                transaction=transaction)
+            roles_list = list([r['role'] for r in res_roles])
+            roles = {}
+            roles["roles"] = roles_list
+            user_roles = {}
+            user_roles[user] = roles
+            users.append(user_roles)
+        return sorted(users)
```

Comment:
I can see why you made `users` a list (because the current return value is a list), but it's better off being a dictionary, because we don't support multiple users with the same name. The format returned here should be:
```
{
  "user1": {
    "roles": ["foo", "bar"]
  },
  "user2": {
    "roles": ["bar"]
  }
}
```

This change should make all of the code that consumes this data cleaner, too.